providers/flagd/src/test/java/dev/openfeature/contrib/providers/flagd/resolver/grpc/GrpcConnectorTest.java

```diff
@@ -1,496 +1,123 @@
-            mockStaticService.when(() -> ServiceGrpc.newBlockingStub(any(Channel.class)))
-                    .thenReturn(mockBlockingStub);
-            mockStaticService.when(() -> ServiceGrpc.newStub(any()))
-                    .thenReturn(mockStub);
-
-            try (MockedStatic<NettyChannelBuilder> mockStaticChannelBuilder = mockStatic(NettyChannelBuilder.class)) {
-
-                mockStaticChannelBuilder.when(() -> NettyChannelBuilder
-                        .forTarget(anyString())).thenReturn(mockChannelBuilder);
-
-                final FlagdOptions flagdOptions = FlagdOptions.builder().host(host).port(port).tls(false).build();
-                new GrpcConnector(flagdOptions, null, null, null);
-
-                // verify host/port matches
-                mockStaticChannelBuilder.verify(() -> NettyChannelBuilder
-                        .forTarget(String.format(targetUri)), times(1));
-            }
-        }
-    }
-
-    @Test
-    void no_args_host_and_port_env_set_should_build_tcp_socket() throws Exception {
-        final String host = "server.com";
-        final int port = 4321;
-        final String targetUri = String.format("%s:%s", host, port);
-
-        new EnvironmentVariables("FLAGD_HOST", host, "FLAGD_PORT", String.valueOf(port)).execute(() -> {
-            ServiceGrpc.ServiceBlockingStub mockBlockingStub = mock(ServiceGrpc.ServiceBlockingStub.class);
-            ServiceGrpc.ServiceStub mockStub = createServiceStubMock();
-            NettyChannelBuilder mockChannelBuilder = getMockChannelBuilderSocket();
-
-            try (MockedStatic<ServiceGrpc> mockStaticService = mockStatic(ServiceGrpc.class)) {
-                mockStaticService.when(() -> ServiceGrpc.newBlockingStub(any(Channel.class)))
-                        .thenReturn(mockBlockingStub);
-                mockStaticService.when(() -> ServiceGrpc.newStub(any()))
-                        .thenReturn(mockStub);
-
-                try (MockedStatic<NettyChannelBuilder> mockStaticChannelBuilder = mockStatic(
-                        NettyChannelBuilder.class)) {
-
-                    mockStaticChannelBuilder.when(() -> NettyChannelBuilder
-                            .forTarget(anyString())).thenReturn(mockChannelBuilder);
-
-                    new GrpcConnector(FlagdOptions.builder().build(), null, null, null);
-
-                    // verify host/port matches & called times(= 1 as we rely on reusable channel)
-                    mockStaticChannelBuilder.verify(() -> NettyChannelBuilder.forTarget(targetUri), times(1));
-                }
-            }
-        });
+    @AfterEach
+    void tearDown() throws Exception {
+        tearDownGrpcServer();
     }
 
-    /**
-     * OS Specific test - This test is valid only on Linux system as it rely on
-     * epoll availability
-     */
-    @Test
-    @EnabledOnOs(OS.LINUX)
-    void path_arg_should_build_domain_socket_with_correct_path() {
```

Comment:
I'm OK with deleting some of these tests if there's no longer any need for them, but I want to be sure that we actually have these covered elsewhere or in e2e. This one in particular is testing unix socket support I'm not sure we have elsewhere.

If there's some reason that this test is super hard to update or maintain, I'm also OK with deleting it here, but maybe we need to have some kind of e2e testing support for unix sockets

WDYT @aepfli ?

Reply:
I think we're mixing integration and unit tests here —in the end, we should test that we're correctly setting everything within the channels, etc. Therefore, I would only need to test our channel builder, which is more like an integration test. To be fair, I am not sure I would cover this with an integration test. 

This is, as you suggest, better suited for an e2e test with the docker image - as sockets are just file mounts, this should be reasonably easy. But I am not sure we need an integration test like this. If we do a proper builder test we should be fine.